server_structs/director.go

```diff
@@ -66,16 +67,17 @@ type (
 	}
 
 	ServerAd struct {
-		Name        string
-		AuthURL     url.URL
-		BrokerURL   url.URL // The URL of the broker service to use for this host.
-		URL         url.URL // This is server's XRootD URL for file transfer
-		WebURL      url.URL // This is server's Web interface and API
-		Type        ServerType
-		Latitude    float64
-		Longitude   float64
-		Writes      bool
-		DirectReads bool // True if reads from the origin are permitted when no cache is available
+		Name         string     `json:"name"`
+		AuthURL      url.URL    `json:"auth_url"`
+		BrokerURL    url.URL    `json:"broker_url"` // The URL of the broker service to use for this host.
+		URL          url.URL    `json:"url"`        // This is server's XRootD URL for file transfer
+		WebURL       url.URL    `json:"web_url"`    // This is server's Web interface and API
+		Type         ServerType `json:"type"`
+		Latitude     float64    `json:"latitude"`
+		Longitude    float64    `json:"longitude"`
+		Writes       bool       `json:"enable_write"`
+		DirectReads  bool       `json:"enable_fallback_read"` // True if reads from the origin are permitted when no cache is available
+		FromTopology bool       `json:"from_topology"`
```

Comment:
It pains me that we have both `from_topology` and `from-topology`, but I understand why it has to be like that 😭

Reply:
and Cannon actually prefers to `fromTopology` #899 😂